### DISCUSSION TITLE
mmdetection에서 wandb를 epoch 기준으로 찍을 수 있나요?

### DISCUSSION THREAD

**Discussion by kmouleejunhyuk** [boostcampaitech2/object-detection-level2-cv-06#44] [Q&A] [answered]
같은 30에포크인데 어떤 모델은 배치가 4라서 스텝이 길어지고, 어떤 모델은 배치가 16이라 그래프에 보여질 떄 x축 비율이 틀어지는 문제가 아래 사진과 같이 발생했습니다.(둘 다 총 30에포크에 20에폭에서 lr이 줄어야 합니다. 파랑선은 배치 16이라 step이 짧고, 초록선은 4라서 step이 상대적으로 김)

![image](https://user-images.githubusercontent.com/49234207/135743312-72ecb51a-79ce-456d-b96a-2a439ebd5801.png)

에포크 기준으로 찍으면 문제가 없을 거 같은데, mmdetection에서 가능할까요?

**Comment by ppskj178** [comment, marked as answer]
by_epoch 옵션으로 설정할 수 있어보이는데
epoch 단위로 안되시는거면
어디선가  False로 설정되어있는지 확인해볼 필요가 있어보입니다.

``` python
...
class WandbLoggerHook(LoggerHook):

    def __init__(self,
                 init_kwargs=None,
                 interval=10,
                 ignore_last=True,
                 reset_flag=False,
                 commit=True,
                 by_epoch=True,
                 with_step=True):
        super(WandbLoggerHook, self).__init__(interval, ignore_last,
                                              reset_flag, by_epoch)
        self.import_wandb()
        self.init_kwargs = init_kwargs
        self.commit = commit
        self.with_step = with_step
...
```

자세한건 공식 문서 참고바랍니다.
https://mmcv.readthedocs.io/en/latest/_modules/mmcv/runner/hooks/logger/wandb.html

**Reply by kmouleejunhyuk** [reply]
@ppskj178 
넵 한번 저 옵션을 만져보겠습니다

**Reply by jiwoo0212** [reply]
준혁님 by_epch = True로하면 되나요?

**Reply by kmouleejunhyuk** [reply]
@jiwoo0212 아직 테스트 전인데 만져보겠습니다 trainval 건드리느라 신경을 못썼네요

**Reply by jiwoo0212** [reply]
@kmouleejunhyuk 오 알겠습니다. 저도 시도해볼게요!
